cmd/topicmappr/commands/rebuildtopics.go

```diff
@@ -197,6 +197,13 @@ func rebuild(cmd *cobra.Command, _ []string) {
 		fmt.Printf("%s[none]\n", indent)
 	}
 
+	// Ignore no-ops in the case of a substitution affinity. As we only care
+	// about broker replacement, the no-ops would inflate the assignment size,
+	// which would put load on the controller for no useful reason.
+	if sa {
```

Comment:
Thoughts about when to skip no-ops:

- During substitution affinity might not be the only time we want to skip no-ops. If we replace 2 brokers with 4 new ones and do not use sub affinity, there will still be a lot of no-ops we might want to skip.
- There's cases that we also might _want_ all the no-ops, such as storing a complete, original map (say, if you ever want to roll back to it or just have as a reference).

Because of these two considerations, it's probably best to specify whether we want to skip no-ops with a flag rather than building it into any specific rebuild pathway.